pyproject.toml

```diff
@@ -3,7 +3,7 @@ name = "atomworks"
 version = "1.0.0"
 description = "A research-oriented data toolkit for training biomolecular deep-learning foundation models"
 readme = "README.md"
-requires-python = ">=3.12"
+requires-python = ">=3.11"
```

Comment:
Just double-checking on this change: Did we run all tests in local CI to verify 3.11 supports everything? 
I remember 3.12 had a few new features & speed-ups and is what we developed & tested with most recently. Also forward-looking, our github CI currently only explicitly checks py3.12.